test/pkg/test/http_gateway.go

```diff
@@ -5,7 +5,7 @@ import (
 	"sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
-func (env *Framework) NewGateway() *v1beta1.Gateway {
+func (env *Framework) NewHttpGateway() *v1beta1.Gateway {
```

Comment:
Is there a reason to make this?  A gateway can have multiple listener(s),  one for HTTP and one for HTTPs.  And need a e2e test case to cover a single HTTPRoute can be accessed through HTTP and HTTPs

Reply:
Maybe put 2 listeners in  one GW is better, I will change to that one in the next revision 			

```
Spec: v1beta1.GatewaySpec{
				GatewayClassName: "amazon-vpc-lattice",
				Listeners: []v1beta1.Listener{
					{
						Name:     "http",
						Protocol: v1beta1.HTTPProtocolType,
						Port:     80,
					},
					{
						Name:     "https",
						Protocol: v1beta1.HTTPSProtocolType,
						Port:     443,
					},
				},
			},
```